src/bluesearch/entrypoint/database/parse.py

```diff
@@ -211,14 +222,30 @@ def run(
         try:
             parsers = iter_parsers(input_type, input_path)
 
-            for parser in parsers:
+            for i, parser in enumerate(parsers):
                 article = Article.parse(parser)
                 output_file = output_dir / f"{article.uid}.json"
 
                 if output_file.exists():
                     raise FileExistsError(f"Output '{output_file}' already exists!")
                 else:
                     serialized = article.to_json()
+
+                    if include_topic:
+                        topic_path = (
+                            input_path.parent.parent
+                            / "topic"
+                            / f"{input_path.stem}.json"
+                        )
+                        topic_json = JSONL.load_jsonl(topic_path)
+
+                        serialized_json = json.loads(serialized)
+                        if input_type == "pubmed-xml-set":
+                            serialized_json["topics"] = topic_json[i]["topics"]
```

Comment:
So the idea here is to dynamically add this `topics` key only inside of the JSON?

Wouldn't it make more sense to define a new attribute `topcis` for `Article` that is an empty list by default? And then modify this attribute once we have the topics and then serialize the whole article?


This way the schema of the `Article` is going to be clear.

Reply:
I like the idea, but I have the feeling that parsing the article and finding the topic are two completely separate steps. I am a bit afraid of making those steps more dependent on each other. Currently, I try to keep everything independent (and the changes in the CI command reasonable). Do you think it is possible to add the new attribute `topics` in `Article` and keeping things independent?

Reply:
I see your point and I guess that is why I suggested to set it to `list[str]` by default. If the user wants, they can provide itlater but at least we declare that the `topics` attribute could be populated.

IMO one should just look at the `Article` dataclass and be able to see what the final schema of the saved `.json` is going to be. Once we start writing code that adds new entries to the `.json` then I am worried about things becoming intractable.

Anyway, what do you think?